packages/main/src/Button.ts

```diff
@@ -363,6 +363,23 @@ class Button extends UI5Element implements IFormElement {
 		this.hasIcon = !!this.icon;
 	}
 
+	async onAfterRendering() {
+		let iconAccessibleName;
+
+		if (!!this.icon && this.iconOnly && !this.tooltip) {
+			const icon = this.shadowRoot!.querySelector("[ui5-icon]") as Icon;
```

Comment:
as a best [practice](https://github.com/SAP/ui5-webcomponents/blob/main/docs/5-development/07-ts.md) (3. Usage of Generics.), we use generic when calling querySelector, `querySelector<Icon>` instead of casting `as Icon`